Handler/Demo.hs

```diff
@@ -0,0 +1,37 @@
+module Handler.Demo where
+
+import Import
+
+import Control.Monad (filterM)
+import Data.Maybe (listToMaybe)
+
+getDemoR :: Handler Html
+getDemoR = do
+    root <- getAppRoot
+    siteId <- runDB $ findDemoSite root
+
+    defaultLayout $ do
+        setTitle "Carnival Demo"
+        $(widgetFile "demo")
+
+findDemoSite :: Text -> DB SiteId
+findDemoSite root = do
+    sites <- selectList [SiteBaseUrl ==. root] []
+    msite <- listToMaybe <$> filterM noMembers sites
+
+    maybe (createDemoSite root) (return . entityKey) msite
+
+  where
```

Comment:
Yeah, I'll add the uniqueness constraint. I had been considering it but hadn't done it yet. Given that, I'll feel better about not enforcing no-members here.